parsers/test/test_CR.py

```diff
@@ -0,0 +1,122 @@
+import json
+from datetime import datetime, timedelta, timezone
+from importlib import resources
+
+from freezegun import freeze_time
+from requests import Session
+from requests_mock import ANY, GET, Adapter
+from snapshottest import TestCase
+
+from parsers.CR import fetch_exchange, fetch_production
+from parsers.lib.exceptions import ParserException
+
+
+class TestCR(TestCase):
+    def setUp(self) -> None:
+        super().setUp()
+
+        self.session = Session()
+        self.adapter = Adapter()
+        self.session.mount("https://", self.adapter)
+
+    @freeze_time("2024-01-01 12:00:00")
+    def test_fetch_production_live(self):
+        """That we can fetch the production mix at the current time."""
+        self.adapter.register_uri(
+            GET,
+            ANY,
+            json=json.loads(
+                resources.files("parsers.test.mocks.CR")
+                .joinpath("production_live.json")
+                .read_text()
+            ),
+        )
+
+        production = fetch_production(session=self.session)
+
+        self.assertMatchSnapshot(
+            [
+                {
+                    "datetime": element["datetime"].isoformat(),
+                    "zoneKey": element["zoneKey"],
+                    "production": element["production"],
+                    "storage": element["storage"],
+                    "source": element["source"],
+                    "sourceType": element["sourceType"].value,
+                    "correctedModes": element["correctedModes"],
+                }
+                for element in production
+            ]
+        )
```

Comment:
FYI I found a "better" snapshot library that both fixes the enum issue and uses pytest style assertions.

We won't be able to use it just yet but hopefully soon.

Reply:
Nice find @VIKTORVAV99 - looks really good! 

I have actually never used snapshot testing libraries before, but from a first look I think it would definitely be worth (thinking of) switching to `syrupy`. If anything, just the fact that it is actively maintained would be a very big plus and I think would prevent bad surprises / major blockers down the line. In addition, as you said, it also seems to be able to do everything that `snapshottest` can do... but "better" 😅  

The docs do say though that the two packages cannot coexist in the same environment because of argument conflicts and overlapping fixtures. Did you already have in mind a possible migration path? 

If you are not too keen for a big-bang migration, one possibility could be to initially keep using `snapshottest` and slowly migrate the existing tests to use `pytest`-style tests and`snapshottest`'s `snapshot` fixture / `snapshot.assert_match()`. This would gradually drop all explicit references to `snapshottest`, which we currently have just to be able to use `snapshottest.TestCase`. Once it's all done, I think it should then be possible to simply swap the `snapshottest` package with `syrupy`, and all tests would still work out-of-the-box: the `snapshot` fixture would this time be the one provided by `syrupy`, and the docs claim that it would understand `snapshottest`'s `snapshot.assert_match(actual)` statement (although it would natively prefer `snapshot == actual`)

Reply:
Honestly I was just going to do one big PR that does it all. All snapshots would change anyway due to a different serialization process for the date times anyway. Since we don’t need to explicitly loop over them and there would be no chance to call isoformat anymore. But perhaps it would just be better to do it incrementally as you mention.

I’m also going to see if we can change it one file at a time without conflicts to see what our options are in regards to that. If that’s an alternative migration would be easy.

I’m going to have a chat with the people in the engineering team tomorrow and see what they say.